streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -1816,6 +1816,102 @@ public void shouldWrapProcessorsForStreamTableJoinWithGracePeriod() {
         assertThat(counter.numConnectedStateStores(), CoreMatchers.is(2));
     }
 
+    @Test
+    public void shouldWrapProcessorsForTableTableInnerJoin() {
+        final Map<Object, Object> props = dummyStreamsConfigMap();
+        props.put(PROCESSOR_WRAPPER_CLASS_CONFIG, RecordingProcessorWrapper.class);
+
+        final WrapperRecorder counter = new WrapperRecorder();
+        props.put(PROCESSOR_WRAPPER_COUNTER_CONFIG, counter);
+
+        final StreamsBuilder builder = new StreamsBuilder(new TopologyConfig(new StreamsConfig(props)));
+
+        final KTable<String, String> t1 = builder.table("input1"); // 1
```

Comment:
nit: use Consumed#as to set the processor name for source tables so you don't have stuff like `KTABLE-SOURCE-0000000002`